### ISSUE TITLE
"Show only window information from current screen" does not work

### ISSUE THREAD

**Issue by 3urobeat** [dhruv8sh/plasma6-window-title-applet#6]
Hey!  
Thank you so much for maintaining this applet for Plasma 6!  

The "Show only window information from current screen" setting sadly does not work for me - it always shows the currently focused application, not matter on which screen it is.  
The behavior does not change on X11.  
Am I missing something?

EndeavourOS on kernel `6.7.8-arch1-1`, running Plasma `6.0.1`. This applet, version `0.3-1`, was installed through the AUR.

**Comment by dhruv8sh** [comment]
Yes. This was reproducable. Thanks for pointing it out!
I unfortunately looked over a bug that does not spit out any error.
It has been fixed, and will now be pushed to the KDE Store as well.

**Comment by dhruv8sh** [comment]
Please review it so I can close this issue.

**Comment by 3urobeat** [comment]
I've cloned the `master` branch, installed the package using `kpackagetool5 -i ./`, relogged and it works now!

Thank you so much for the fast fix \^\^

**Comment by D3SOX** [comment]
Maybe we could reach out to KDE developers and ask them to implement a way to track the last active window per screen? I think they're active on Matrix.

**Comment by 3urobeat** [comment]
> Plasma Frameworks currently does not provide a way to keep track of a top window in a desktop

Yeah, that was what I feared in the first place when making this suggestion.  
Sure, we could try to ask the KDE/Plasma team but I'm not sure exactly how and if they'd even be open to adding this on the fly.  

We also can't forget about the restrictions Wayland intentionally puts up to prevent applications from sniffing on one another (although I'm not sure how relevant that is in the context of a Plasma Applet).  

I don't exactly know where to go from here. Remembering the last focused application name on a screen could theoretically work. Is there an event fired when an application is closed? If so we could listen for that and update the title accordingly.  
But I'm just poking around in the dark here as I have never worked on an applet myself.

**Comment by ChrTall** [comment]
I also use applet-window-buttons which was ported by moodyhunter to KF6.
[https://github.com/moodyhunter/applet-window-buttons6](https://github.com/moodyhunter/applet-window-buttons6)

It has a Setting **per-screen-active-window** which does exactly what is discussed here.
The applet does not loose track of the window even if you focus another window on a different screen.
![applet-window-buttons-settings](https://github.com/dhruv8sh/plasma6-window-title-applet/assets/55497425/9862265f-6b53-49ce-bf82-2067f85220fb)

Maybe you can get some inspiration from there.
Anyways great work, thank you.

**Comment by dhruv8sh** [comment]
![2024-05-11 17-25-51](https://github.com/dhruv8sh/plasma6-window-title-applet/assets/67322047/0be85e98-0528-46f8-9f58-70e5ff8b11b4)

The method used in this widget that you mention is sticky. I have this implemented locally already. Problem with it is that it would stay active even when I minimize, the app's name is shown.

**Comment by ChrTall** [comment]
I tested the behaviour in your video with the applet-window-buttons6 applet and I do not get the same behaviour.
It disappears, when minimizing the window with the buttons from the applet or the native buttons from the window.
Maybe the logic in the other applet is a bit different, but I did not really dig deep.
Maybe you can give the other applet-window-buttons6  a try, when you have some time.
